docs/react-types/ComponentProps.md

```diff
@@ -0,0 +1,57 @@
+---
+title: ComponentProps<T>
+---
+
+`ComponentProps<T>` is a utility type that lets you grab all valid props of an HTML or SVG element, or infer prop type for a component.
```

Comment:
It's for any element type. HTML props can be used as an example but it makes it sound like these are the intended target of this type when they're just examples.